### PR TITLE
JP-2075: fixed flux scaling issues in wfss_contam

### LINKED ISSUE #6016: Investigate scaling issues of simulated spectra in wfss_contam step
_Issue [JP-2075](https://jira.stsci.edu/browse/JP-2075) was created on JIRA by [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse):_

Testing of the prototype wfss_contam step using the latest NIRCam WFSS MIRAGE simulations shows a difference in the overall signal level of simulated contamination spectra generated by the step relative to the spectra in the grism image being corrected. Check for proper application of flux calibration terms and possible differences in flux calibration data that were used in MIRAGE vs the current CRDS ref files.

### PR DESCRIPTION
**Pull request by emolter**
Resolves [JP-2075](https://jira.stsci.edu/browse/JP-2075)

Closes #6016 

This PR fixes a difference in the overall signal level of simulated contamination spectra generated by the step relative to the spectra in the grism image being corrected.  The flux scaling issues were found to be caused by a combination of two problems:  First, background fluxes were being propagated into the grism image when the input direct image was not background subtracted, leading to unphysical broadening in the cross-dispersion direction for the model dispersed sources.  Second, flux oversampling in the dispersed_pixel() function was leading to double-counting of flux; dividing the result by the oversampling factor resolved this.

**Checklist for maintainers**
- [x] added entry in `CHANGES.rst` within the relevant release section
- [x] updated or added relevant tests
- [ ] updated relevant documentation
- [x] added relevant milestone
- [x] added relevant label(s)
- [x] ran regression tests, post a link to the Jenkins job below.
      [How to run regression tests on a PR](https://github.com/spacetelescope/jwst/wiki/Running-Regression-Tests-Against-PR-Branches)
- [ ] Make sure the JIRA ticket is [resolved properly](https://github.com/spacetelescope/jwst/wiki/How-to-resolve-JIRA-issues)

### CODE REVIEW

File: jwst/wfss_contam/observations.py [hunk 1]
```diff
@@ -10,10 +10,53 @@
...
+    -----
+    Improper background subtraction in input _i2d image leads to extra flux
+    in the simulated dispersed image, and was one cause of flux scaling issues
+    in a previous version.
+    """
+    if box_size is None:
+        box_size = (int(data.shape[0]/5), int(data.shape[1]/5))
+    sigma_clip = SigmaClip(sigma=sigma)
+    bkg_estimator = MedianBackground()
+    bkg = Background2D(data, box_size, filter_size=filter_size,
```
Comment (hbushouse) [inline jwst/wfss_contam/observations.py:53]: Is the `data` array used here the full i2d image array?  Just curious.

File: jwst/wfss_contam/tests/test_observations.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+@pytest.fixture(scope='module')
+def direct_image_with_gradient(tmp_cwd_module, direct_image):
+    ny, nx = direct_image.shape
+    y, x = np.mgrid[:ny, :nx]
+    gradient = x * y / 5000.0
+    data = direct_image + gradient
+
+    # obs expects input list of direct image filenames
+    model = ImageModel(data=data)
+    model.save(DIR_IMAGE)
```
Comment (hbushouse) [inline jwst/wfss_contam/tests/test_observations.py:38]: Thanks for creating the image on the fly, because we try to avoid adding data files to the repo.

**Review by hbushouse — COMMENTED** [review]
Looks great. Just a couple minor questions.

**Comment by hbushouse** [comment]
Have you run a regression test? I believe we have 1 or 2 tests of wfss_contam in the regtest suite. Of course we'll expect differences/failures due to these updates.

File: jwst/wfss_contam/observations.py (hunk 1 above)
Comment (emolter) [reply jwst/wfss_contam/observations.py:53]: yes, that's correct

File: jwst/wfss_contam/tests/test_observations.py (hunk 2 above)
Comment (emolter) [reply jwst/wfss_contam/tests/test_observations.py:38]: yeah I figured that's what you would want

**Comment by emolter** [comment]
Didn't realize there were any wfss_contam regtests; run started [here](https://plwishmaster.stsci.edu:8081/job/RT/job/JWST-Developers-Pull-Requests/1419/)

**Comment by hbushouse** [comment]
> Didn't realize there were any wfss_contam regtests; run started [here](https://plwishmaster.stsci.edu:8081/job/RT/job/JWST-Developers-Pull-Requests/1419/)

Test results show zero failures, which normally is a good thing, but I expected differences, because we do have 1 regtest for this: test_nircam_wfss_contam.py.  However, I see that that test module is set to be skipped, because of the ridiculously long run time. Can you try running that one test locally in your PR branch environment?

**Comment by emolter** [comment]
sure, started that just now

**Review by hbushouse — APPROVED** [review]
Looks great!

**Pull request merged**